packages/web-app-files/src/helpers/share/link.ts

```diff
@@ -16,9 +22,20 @@ interface CreateQuicklink {
 
 export const createQuicklink = async (args: CreateQuicklink): Promise<Share> => {
   const { resource, store, password, $gettext, ability } = args
+
+  const allowResharing = store.state.user.capabilities.files_sharing?.resharing
+  const isDefaultRoleViewer = store.state.user.capabilities.files_sharing?.quickLink?.default_role
+  const internalPerm = (resource.isFolder ? linkRoleInternalFolder : linkRoleInternalFile).bitmask(
+    allowResharing
+  )
+  const viewerPerm = (resource.isFolder ? linkRoleViewerFolder : linkRoleViewerFile).bitmask(
+    allowResharing
+  )
+  const canCreatePublicLink = ability.can('create-all', 'PublicLink')
   const params: { [key: string]: unknown } = {
     name: $gettext('Quicklink'),
-    permissions: ability.can('create-all', 'PublicLink') ? '1' : '0',
+    permissions:
+      canCreatePublicLink && isDefaultRoleViewer === 'viewer' ? viewerPerm : internalPerm,
```

Comment:
This is still not quite right I think. `isDefaultRoleViewer` (= `defaultRole`, or?) should only apply if it's set (and the permissions grant it). If it's not set, we default to the viewer role (if permissions grant it - otherwise internal).

Reply:
@JammingBen should be correct now, if you may check and confirm :)